plugins/styleguide/assets/javascripts/discourse/components/sections/organisms/modal.hbs

```diff
@@ -1,15 +1,57 @@
+{{! template-lint-disable no-potential-path-strings}}
+
 <StyleguideExample @title="<DModal>">
-  <DModal
-    @closeModal={{@dummyAction}}
-    @modalStyle="inline-modal"
-    @title={{i18n "styleguide.sections.modal.header"}}
-  >
-    <DModalBody>
-      {{html-safe @dummy.lorem}}
-    </DModalBody>
+  <Styleguide::Component>
+    <DModal
+      @closeModal={{fn (mut this.inline) true}}
+      @inline={{this.inline}}
+      @title={{this.title}}
+      @subtitle={{this.subtitle}}
+      @flash={{this.flash}}
+      @flashType={{this.flashType}}
+      @errors={{this.errors}}
+      @dismissable={{this.dismissable}}
+    >
+      <:body>
+        {{this.body}}
+      </:body>
+
+      <:footer>
+        {{i18n "styleguide.sections.modal.footer"}}
+      </:footer>
+    </DModal>
+  </Styleguide::Component>
 
-    <div class="modal-footer">
-      {{i18n "styleguide.sections.modal.footer"}}
-    </div>
-  </DModal>
+  <Styleguide::Controls>
+    <Styleguide::Controls::Row @name="@inline">
+      <DToggleSwitch @state={{this.inline}} {{on "click" this.toggleInline}} />
+    </Styleguide::Controls::Row>
+    <Styleguide::Controls::Row @name="@dismissable">
+      <DToggleSwitch
+        @state={{this.dismissable}}
+        {{on "click" this.toggleDismissable}}
+      />
+    </Styleguide::Controls::Row>
+    <Styleguide::Controls::Row @name="@title">
+      <Input @value={{this.title}} />
+    </Styleguide::Controls::Row>
+    <Styleguide::Controls::Row @name="@subtitle">
+      <Input @value={{this.subtitle}} />
+    </Styleguide::Controls::Row>
+    <Styleguide::Controls::Row @name="<:body>">
+      <Textarea @value={{this.body}} />
+    </Styleguide::Controls::Row>
+    <Styleguide::Controls::Row @name="@flash">
+      <Input @value={{this.flash}} />
+    </Styleguide::Controls::Row>
+    <Styleguide::Controls::Row @name="@flashType">
+      <ComboBox
+        @value={{this.flashType}}
+        @content={{this.flashTypes}}
+        @onChange={{fn (mut this.flashType)}}
+        @valueProperty={{null}}
+        @nameProperty={{null}}
+      />
+    </Styleguide::Controls::Row>
+  </Styleguide::Controls>
```

Comment:
Taking advantage of the work @jjaffeux did recently on interactive styleguide components

<img width="703" alt="SCR-20230619-kznw" src="https://github.com/discourse/discourse/assets/6270921/5c319bda-4626-470d-9a2f-4ed9e1dde31d">